### PR TITLE
Alter RuntimeComponentManifest.targets to use the correct runtime pack

### LINKED ISSUE #56299: Mono: The logic to find the RuntimeComponentManifest.json file is incorrect
### Description

The logic to find the RuntimeComponentManifest.json here:

https://github.com/dotnet/runtime/blob/893739ffe77df0eb927d3755e08647597dae32a6/src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/RuntimeComponentManifest.targets#L6

does not take into account that `ResolvedRuntimePack` might contain more than one runtime pack.

See binlog: 
[msbuild.binlog.zip](https://github.com/dotnet/runtime/files/6878265/msbuild.binlog.zip)

Here's the `ResolvedRuntimePack` items:

```
ResolvedRuntimePack
    Microsoft.NETCore.App.Runtime.Mono.osx-x64
        FrameworkName = Microsoft.NETCore.App
        NuGetPackageVersion = 6.0.0-rc.1.21371.1
        PackageDirectory = /Users/rolf/work/maccore/dotnet-tools/xamarin-macios/tests/dotnet/packages/microsoft.netcore.app.runtime.mono.osx-x64/6.0.0-rc.1.21371.1
        RuntimeIdentifier = osx-x64
        NuGetPackageId = Microsoft.NETCore.App.Runtime.Mono.osx-x64
        IsTrimmable = 
        OriginalItemSpec = Microsoft.NETCore.App.Runtime.Mono.osx-x64
    Microsoft.AspNetCore.App.Runtime.osx-x64
        FrameworkName = Microsoft.AspNetCore.App
        NuGetPackageVersion = 6.0.0-rc.1.21371.15
        PackageDirectory = /Users/rolf/work/maccore/dotnet-tools/xamarin-macios/tests/dotnet/packages/microsoft.aspnetcore.app.runtime.osx-x64/6.0.0-rc.1.21371.15
        RuntimeIdentifier = osx-x64
        NuGetPackageId = Microsoft.AspNetCore.App.Runtime.osx-x64
        IsTrimmable = 
        OriginalItemSpec = Microsoft.AspNetCore.App.Runtime.osx-x64
    Microsoft.macOS.Runtime.osx-x64
        FrameworkName = Microsoft.macOS
        NuGetPackageVersion = 12.0.100-ci.dotnet-pinvoke-lookup.287
        PackageDirectory = /Users/rolf/work/maccore/dotnet-tools/xamarin-macios/tests/dotnet/packages/microsoft.macos.runtime.osx-x64/12.0.100-ci.dotnet-pinvoke-lookup.287
        RuntimeIdentifier = osx-x64
        NuGetPackageId = Microsoft.macOS.Runtime.osx-x64
        IsTrimmable = 
        OriginalItemSpec = Microsoft.macOS.Runtime.osx-x64
```

The end result is that the last one is picked, and the code fails because it expects to find RuntimeComponentManifest.json in `Microsoft.macOS.Runtime.osx-x64`, and not `Microsoft.NETCore.App.Runtime.Mono.osx-x64`.

CC @lambdageek, @steveisok

### PR DESCRIPTION
**Pull request by steveisok**
When trying to process what runtime components are available, we were assuming %(ResolvedRuntimePack.PackageDirectory) would give us the correct Microsoft.NETCore.App runtime pack.  Unfortunately, the ordering was likely altered when we started setting <KnownRuntimePack> items in the Mono workload.  Since relying on position of the items isn't reliable, we will instead use %(ResolvedFrameworkReference) to tell us which runtime pack is active and where it's located.

Fixes https://github.com/dotnet/runtime/issues/56299

### CODE REVIEW

**Review by lambdageek — APPROVED** [review]
Should we have a check somewhere that `@(ResolvedFrameworkReference->WithMetadataValue('Identity', 'Microsoft.NETCore.App')->Count()) == 1` ?

**Review by rolfbjarne — APPROVED** [review]
If I apply this fix locally, it works.

**Comment by radical** [comment]
Wasm workload tests are failing with:

```
 The "MonoRuntimeComponentManifestReadTask" task failed unexpectedly. [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
 System.IO.DirectoryNotFoundException: Could not find a part of the path '/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/ [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
         /datadisks/disk1/work/B43109B2/p/dotnet-workload/packs/Microsoft.NETCore.App.Runtime.Mono.browser-wasm/6.0.0-ci\runtimes\browser-wasm\build\RuntimeComponentManifest.json [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
       '. [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode) in System.Private.CoreLib.dll:token 0x60000d5+0x0 [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) in System.Private.CoreLib.dll:token 0x60000d9+0x31 [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) in System.Private.CoreLib.dll:token 0x6006142+0x78 [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
    at System.IO.File.OpenRead(String path) in System.Private.CoreLib.dll:token 0x6005cbb+0x9 [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
    at JsonToItemsTaskFactory.JsonToItemsTaskFactory.JsonToItemsTask.TryGetJson(String jsonFilePath, JsonModelRoot& json) in JsonToItemsTaskFactory.dll:token 0x6000019+0x2 [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
    at JsonToItemsTaskFactory.JsonToItemsTaskFactory.JsonToItemsTask.Execute() in JsonToItemsTaskFactory.dll:token 0x6000018+0x1f [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() in Microsoft.Build.dll:token 0x60015fb+0x3e [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) in Microsoft.Build.dll:token 0x600147a+0x804 [/datadisks/disk1/work/B43109B2/w/A32D0938/e/qql1xr5s.l1p/rebuild_Release_True.csproj]
```

And that seems to be because the json path is:
```
                   Set Property: _MonoRuntimeComponentManifestJsonFilePath=
                           /datadisks/disk1/work/B43109B2/p/dotnet-workload/packs/Microsoft.NETCore.App.Runtime.Mono.browser-wasm/6.0.0-ci\runtimes\browser-wasm\build\RuntimeComponentManifest.json

                   Initializing task factory "JsonToItemsTaskFactory.JsonToItemsTaskFactory" from assembly "/datadisks/disk1/work/B43109B2/p/dotnet-workload/packs/Microsoft.NET.Runtime.MonoTargets.Sdk/6.0.0-ci/Sdk/../tasks/net6.0/JsonToItemsTaskFactory.dll".
```

Note the newlines in the path.

fix:
1. don't use newlines when setting the path
2. The task should check the path, and log an error

File: src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/RuntimeComponentManifest.targets [hunk 1]
```diff
@@ -3,7 +3,10 @@
     <Error Condition="'%(ResolvedRuntimePack.PackageDirectory)' == '' and '$(_MonoRuntimeComponentManifestJsonFilePath)' == ''"
            Text="Empty ResolvedRuntimePack.PackageDirectory while trying to read runtime components manifest" />
     <PropertyGroup>
-      <_MonoRuntimeComponentManifestJsonFilePath Condition="'$(_MonoRuntimeComponentManifestJsonFilePath)' == ''">%(ResolvedRuntimePack.PackageDirectory)\runtimes\$(RuntimeIdentifier)\build\RuntimeComponentManifest.json</_MonoRuntimeComponentManifestJsonFilePath>
+      <_MonoRuntimeComponentManifestJsonFilePath 
+        Condition="'$(_MonoRuntimeComponentManifestJsonFilePath)' == '' and '%(ResolvedFrameworkReference.Identity)' == 'Microsoft.NETCore.App'">
```
Comment (radical) [inline src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/RuntimeComponentManifest.targets:7]: Besides the new line fix, also check if `'%(RuntimePackPath)' != ''`

File: src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/RuntimeComponentManifest.targets [hunk 2]
```diff
@@ -3,7 +3,10 @@
...
+        %(RuntimePackPath)\runtimes\$(RuntimeIdentifier)\build\RuntimeComponentManifest.json
+      </_MonoRuntimeComponentManifestJsonFilePath>
     </PropertyGroup>
```
Comment (radical) [inline src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/RuntimeComponentManifest.targets:10]: And we can error out here, if `$(_MonoRuntimeComponentManifestJsonFilePath)` does not exist.

**Review by lambdageek — APPROVED** [review]
:shipit:

File: src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/RuntimeComponentManifest.targets [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
+           Text="Empty ResolveFrameworkReference.RuntimePackPath while trying to read runtime components manifest" />
+
     <PropertyGroup>
-      <_MonoRuntimeComponentManifestJsonFilePath Condition="'$(_MonoRuntimeComponentManifestJsonFilePath)' == ''">%(ResolvedRuntimePack.PackageDirectory)\runtimes\$(RuntimeIdentifier)\build\RuntimeComponentManifest.json</_MonoRuntimeComponentManifestJsonFilePath>
+      <_MonoRuntimeComponentManifestJsonFilePath 
+        Condition="'$(_MonoRuntimeComponentManifestJsonFilePath)' == '' and '%(ResolvedFrameworkReference.Identity)' == 'Microsoft.NETCore.App'">$([MSBuild]::NormalizePath('%(RuntimePackPath)', 'runtimes', '$(RuntimeIdentifier)', 'build', 'RuntimeComponentManifest.json'))</_MonoRuntimeComponentManifestJsonFilePath>
     </PropertyGroup>
+
+    <Error Condition="!Exists('$(_MonoRuntimeComponentManifestJsonFilePath)')"
+           Text="Unable to read the runtime components manifest as the path does not exist" />
```
Comment (radical) [inline src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/RuntimeComponentManifest.targets:12]: nit: having the path in the error message would be useful.

Comment (steveisok) [reply src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/RuntimeComponentManifest.targets:12]: Keeping me honest ;-)

**Review by radical — APPROVED** [review]

**Comment by radical** [comment]
wasm Workload tests failing with:

`/datadisks/disk1/work/AB8D0934/p/dotnet-workload/packs/Microsoft.NET.Runtime.MonoTargets.Sdk/6.0.0-ci/Sdk/RuntimeComponentManifest.targets(3,5): error : Empty ResolveFrameworkReference.RuntimePackPath while trying to read runtime components manifest`

I'll try this out locally.

**Pull request merged**